ziti-builder.sh

```diff
@@ -0,0 +1,130 @@
+        git config --file "$GIT_CONFIG_GLOBAL" --add safe.directory "${SAFE}"
+    done
+}
+
+function set_workspace(){
+    # let GitHub Actions override the workspace dir
+    if [[ -n "${GITHUB_WORKSPACE:-}" ]]; then
+        WORKDIR="${GITHUB_WORKSPACE}"
+    else
+        export WORKDIR="/github/workspace"
+    fi
+
+    # if project is mounted on WORKDIR then build, else restart in container
+    if [[ -x "${WORKDIR}/${BASENAME}" ]]; then
+        # container environment defines BUILD_ENVIRONMENT=ziti-cmake-docker
+        if [[ "${BUILD_ENVIRONMENT:-}" == "ziti-cmake-docker" ]]; then
+            echo "INFO: running in ziti-cmake container"
+            set_git_safe_dirs "${WORKDIR}"
+        else
+            echo "ERROR: not running in ziti-cmake container" >&2
+            exit 1
+        fi
+    else
+        echo -e "INFO: project not mounted on ${WORKDIR}, re-running in container"\
+            "\nINFO: 'docker run --user ${UID} --volume ${BASEDIR}:${WORKDIR} openziti/ziti-cmake ${WORKDIR}/${BASENAME} ${*}'"
+        exec docker run \
+            --rm \
+            --user "${UID}" \
+            --volume "${BASEDIR}:${WORKDIR}" \
+            --platform "linux/amd64" \
+            openziti/ziti-cmake \
+                "${WORKDIR}/${BASENAME}" "${@}"
+    fi
+}
+
+function main() {
+    echo "INFO: GIT_DISCOVERY_ACROSS_FILESYSTEM=${GIT_DISCOVERY_ACROSS_FILESYSTEM:-}"
+    echo "INFO: WORKDIR=${PWD}"
+    echo "INFO: $(git --version)"
+    echo "INFO: GIT_CONFIG_GLOBAL=${GIT_CONFIG_GLOBAL:-}"
+    # use this value to detect whether any options were passed so we can warn if
+    # they're being ignored when an override command is sent at the same time
+    : "${OPTS:=0}"
+
+    while getopts 'c::p:t:' OPT; do
+        case "${OPT}" in
+            c)  CMAKE_CONFIG="${OPTARG}"
+                OPTS=1
+            ;;
+            p)  CMAKE_PRESET="${OPTARG}"
+                OPTS=1
+            ;;
+            t)  CMAKE_TARGET="${OPTARG}"
+                OPTS=1
+                if [[ ${CMAKE_TARGET} == package ]]; then
+                    BUILD_DIST_PACKAGES=ON
+                fi
+            ;;
+            *)  echo "ERROR: invalid option: ${OPT}" >&2
+                return 1
+            ;;
+        esac
+    done
+    shift $((OPTIND-1))
+    # if args then run them, else clean build output dir and build default target
+    if (( $# )); then
+        if (( OPTS )); then
+            echo "WARN: ignoring options because override command is present" >&2
+        fi
+        exec "${@}"
+    else
+        [[ -d ./build ]] && rm -rf ./build
+        cmake \
+            -E make_directory \
+            ./build  
+        cmake \
+            --preset "${CMAKE_PRESET:-ci-linux-x64}" \
+            -DCMAKE_BUILD_TYPE="${CMAKE_CONFIG:-Release}" \
+            -DBUILD_DIST_PACKAGES="${BUILD_DIST_PACKAGES:-OFF}" \
+            -DVCPKG_OVERLAY_PORTS="./vcpkg-overlays/linux-syslibs/ubuntu18" \
```

Comment:
This script always runs on Ubuntu 18.04 because that's the source image for the builder image `ziti-builder`, and this script always builds the "bundle" target, not "package," when running in CI because we have the distro-specific builder images for target "package."

I was thinking that a dev might want to build target "package" locally, so I left that option in the script. Does it answer your concern to know that package artifacts built with this script are never released? If so, should we make the script more flexible to use the default preset for local builds of target "package" or "bundle" or both?